test/server/repositories/UrlRepository.test.ts

```diff
@@ -323,6 +324,46 @@ describe('UrlRepository', () => {
     })
   })
 
+  describe('getLongUrlAndDescription', () => {
+    it('should return from db when cache is empty', async () => {
+      await expect(repository.getLongUrlAndDescription('a')).resolves.toEqual({
+        longUrl: 'aa',
+        description: 'bb',
+      })
+    })
+
+    it('should return from cache when cache is filled', async () => {
+      const longUrlAndDescription = { longUrl: 'aaa', description: 'bbb' }
+      redisMockClient.set('a', JSON.stringify(longUrlAndDescription))
+      await expect(repository.getLongUrlAndDescription('a')).resolves.toEqual(
+        longUrlAndDescription,
+      )
+    })
+
+    it('should return from db when cache is down', async () => {
+      cacheGetSpy.mockImplementationOnce((_, callback) => {
+        if (!callback) {
+          return false
+        }
+        callback(new Error('Cache down'), 'Error')
+        return false
+      })
+      await expect(repository.getLongUrlAndDescription('a')).resolves.toEqual({
+        longUrl: 'aa',
+        description: 'bb',
+      })
+    })
+
+    it('should return from db when cache value is in the old format (backwards compatible)', async () => {
+      // old format - key: short url, value: long url
+      redisMockClient.set('a', 'aa')
+      const longUrlAndDescription = { longUrl: 'aa', description: 'bb' }
+      await expect(repository.getLongUrlAndDescription('a')).resolves.toEqual(
+        longUrlAndDescription,
+      )
```

Comment:
Can you verify that the database is called?

Reply:
updated!